src/test/java/org/retest/rebazer/service/BitbucketServiceTest.java

```diff
@@ -53,4 +63,118 @@ public void hasChangedSinceLastRun_should_return_true_if_pullrequest_did_change(
 		assertThat(cut.hasChangedSinceLastRun(pr)).isTrue();
 	}
 
+	@Test
+	public void rebaseNeeded_should_return_false_if_headOfBranch_is_equal_to_lastCommonCommitId() {
+		PullRequest pullRequest = mock(PullRequest.class);
+		BitbucketService cut = mock(BitbucketService.class);
+		String head = "12325345923759135";
+		when(cut.getHeadOfBranch(pullRequest)).thenReturn(head);
+		when(cut.getLastCommonCommitId(pullRequest)).thenReturn(head);
+		when(cut.rebaseNeeded(pullRequest)).thenCallRealMethod();
+
+		assertThat(cut.rebaseNeeded(pullRequest)).isFalse();
+	}
+
+	@Test
+	public void rebaseNeeded_should_return_true_if_headOfBranch_isnt_equal_to_lastCommonCommitId() {
+		PullRequest pullRequest = mock(PullRequest.class);
+		BitbucketService cut = mock(BitbucketService.class);
+		String head = "12325345923759135";
+		String lcci = "21342343253253452";
+		when(cut.getHeadOfBranch(pullRequest)).thenReturn(head);
+		when(cut.getLastCommonCommitId(pullRequest)).thenReturn(lcci);
+		when(cut.rebaseNeeded(pullRequest)).thenCallRealMethod();
+
+		assertThat(cut.rebaseNeeded(pullRequest)).isTrue();
+	}
+
+	@Test
+	public void isApproved_should_return_false_if_approved_is_false() {
+		PullRequest pullRequest = mock(PullRequest.class);
+		String json = "{participants: [{\"approved\": false}]}\"";
+		when(bitbucketTemplate.getForObject(anyString(), eq(String.class))).thenReturn(json);
+
+		assertThat(cut.isApproved(pullRequest)).isFalse();
+	}
+
+	@Test
+	public void isApproved_should_return_ture_if_approved_is_true() {
+		PullRequest pullRequest = mock(PullRequest.class);
+		String json = "{participants: [{\"approved\": true}]}\"";
+		when(bitbucketTemplate.getForObject(anyString(), eq(String.class))).thenReturn(json);
+
+		assertThat(cut.isApproved(pullRequest)).isTrue();
+	}
+
+	@Test
+	public void greenBuildExists_should_return_false_if_state_is_failed() {
+		PullRequest pullRequest = mock(PullRequest.class);
+		String json = "{values: [{\"state\": FAILED}]}";
+		when(bitbucketTemplate.getForObject(anyString(), eq(String.class))).thenReturn(json);
+
+		assertThat(cut.greenBuildExists(pullRequest)).isFalse();
+	}
+
+	@Test
+	public void greenBuildExists_should_return_true_if_state_is_successful() {
+		PullRequest pullRequest = mock(PullRequest.class);
+		String json = "{values: [{\"state\": SUCCESSFUL}]}";
+		when(bitbucketTemplate.getForObject(anyString(), eq(String.class))).thenReturn(json);
+
+		assertThat(cut.greenBuildExists(pullRequest)).isTrue();
+	}
+
+	@Test
+	public void getAllPullRequestIds_should_return_false_if_list_isnt_equal() {
+		Repository repo = mock(Repository.class);
+		String json = "{\r\n" + "    \"values\": \r\n" + "    [\r\n" + "        {\r\n" + "            \"id\": 2,\r\n"
```

Comment:
Save this (and the JSON below) under `src/test/resources` and load them in the test.